packages/select/src/ResourceSelect.js

```diff
@@ -163,8 +164,29 @@ const ResourceSelect = ({
       };
     }
     if (onPageChange) onPageChange(inputValue, page);
+
+    // if the UI is filtering, all the options should be present already
+    if (
+      shouldFilterByAllOptions &&
+      !hasMore &&
+      inputValue.length > 0 &&
+      (inputValue.length >= minCharsToSearch || minCharsToSearch === undefined)
```

Comment:
i think we can safely ignore `minCharsToSearch` here. `minCharsToSearch` was introduced to reduce load on our APIs. if we're filtering client side, i don't see the need.